packages/fx-core/src/plugins/resource/bot/appStudio/appStudio.ts

```diff
@@ -161,11 +161,23 @@ export class AppStudio {
     let response = undefined;
     try {
       if (registration.botId) {
-        const getBotRegistrationResponse = await RetryHandler.Retry(
-          () => axiosInstance.get(`${AppStudio.baseUrl}/api/botframework/${registration.botId}`),
+        const getBotRegistrationResponse: AxiosResponse<any> | undefined = await RetryHandler.Retry(
+          async () => {
+            try {
+              return await axiosInstance.get(
+                `${AppStudio.baseUrl}/api/botframework/${registration.botId}`
+              );
+            } catch (e) {
+              if (e.response?.status === 404) {
```

Comment:
Do we also need to include other 4xx user error?

Reply:
I've thought about it after previous discussion with you, but think it may be not worth the efforts to include more 4xx, because:
1. As you can see from the REST api, it's only about `Get xxx/api/{botId}` and `botId` would be a good since there's a check before. My point is there's no big chance for other 4xx except 404.
2. Think from the other way, if we include other 4xx, then the logic may be throwing some error messages for other 4XX to the end users. It may iterate all 4xx and give different error messages, kind of complex.